### PR TITLE
[IGNORE : OUTDATED] Default Xpring Client

### PR DESCRIPTION
**Pull request by dangell7**

### CODE REVIEW

**Comment by dangell7** [comment]
@keefertaylor Its not exactly what you wanted. I moved everything to the legacy and created new clients. Still trying to get the signing done. Now it returns undefined.

**Comment by dangell7** [comment]
I traced the error all the way back to here

```
/// Serialize a transaction to a JavaScript object.
    ///
    /// - Parameter transaction: The transaction to serialize.
    /// - Returns: A JSValue representing the transaction.
    public func serialize(transaction: Rpc_V1_Transaction) -> JSValue? {
        guard
            let transactionData = try? transaction.serializedData()
        else {
            return nil
        }
        let transactionBytes = [UInt8](transactionData)
        print(deserializeTransactionFunction.call(withArguments: [transactionBytes])!)
        return deserializeTransactionFunction.call(withArguments: [transactionBytes])!
    }
```

the print output is undefined where the legacy is:

```
XVnJMYQFqA8EAijpKh5EdjEY5JqyxykMKKSbrUX8uchF6U8,12,40,1,,rsegqrgSP8XmhCYwL9enkZ9BNDNawfPZnn,02FD0E8479CE8182ABD35157BB0FA17A469AF27DCB12B5DDED697C61809116A33B
```

**Comment by dangell7** [comment]
@keefertaylor I'm getting an error when trying to get the accountBalance. The error in the client is `account not found` and the error in the ripple console is `Can't get seq 0 from 3760759 error`

**Comment by keefertaylor** [comment]
Sorry if I'm missing comments on other threads or if we duplicated work. Last time I had looked at your pending work I hadn't seen the existing work refactored. 

You should be able to merge in my legacy code changes with:
```
$ git fetch origin
$ git merge origin/master
```

I'll review this PR shortly.

**Comment by dangell7** [comment]
@keefertaylor No way man, I'm following you and I don't mind the extra work at all. I def didn't reactor the legacy, I just pulled your pr about halfway through day yesterday. Like I said, I couldn't figure out how you wanted the client to work with the legacy so if you want to open my pr and adjust the client thats cool. I just refactored EVERYTHING into legacy documents. It seems to be working, sorta, I have to get the ripple server installed on a production server to test. It seems my dev env wasn't fast enough. Also, because I'm new at this, I may delete pr's, if its deleted, just ignore any comments. Sorry for the confusion on that.

**Comment by dangell7** [comment]
@keefertaylor Can you help me with the server side of the dev rippled server? I have it running but I cant figure out how to open the port? The server is at `142.93.73.197`.  I did `ufw allow 50051`. I was under the impression that this would work... Do I need to add something?

**Comment by dangell7** [comment]
@keefertaylor is this the right track? Seems good. https://www.nginx.com/blog/nginx-1-13-10-grpc/

**Comment by dangell7** [comment]
Yeah that did it.

**Comment by dangell7** [comment]
@keefertaylor Still getting the signing error. I made a commit to fix the get balance function. Aparently I needed to submit the classic address and not the x address to get the balance. Also, the getTx function is failing. The error I receive is `failed - Failed retrieving transaction hash with error: callError(successful, status notFound: txn not found` I've checked the tx on both the test rippled net and my own private test net and the tx is there. 

The getTx for your reference. I think the error is in that I send data and not the hex string?

```
/// Retrieve the raw transaction data for the given transaction hash.
    ///
    /// - Parameter transactionHash: The hash of the transaction.
    /// - Throws: An error if there was a problem communicating with the XRP Ledger.
    /// - Returns: The data of the given transaction.
    public func getRawTx(for transactionHash: TransactionHash) throws -> Rpc_V1_GetTxResponse {
        let transactionRequest = Rpc_V1_GetTxRequest.with {
            $0.hash = Data(transactionHash.utf8)
        }
        return try networkClient.getTx(transactionRequest)
    }
```

**Comment by keefertaylor** [comment]
> Data(transactionHash.utf8)
This is tricky. When we hex encode something, we represent it in ASCII. For instance, the bytes:
`1010 1011 1100 1101`
encode to:
`ABCD`
The utf8 values of `ABCD` is (in base 10) `41, 42, 43, 44` which in turn translates to different bytes than the input. 

There are some helpful routines here:
https://stackoverflow.com/questions/42731023/how-do-i-convert-hexstring-to-bytearray-in-swift-3/42731436

A hex to bytes method will likely be needed at some point. This could get implemented on the `Utils` class.

File: XpringKit/DefaultXpringClient.swift [hunk 1]
```diff
@@ -1,146 +1,181 @@
...
+    /// - Throws: An error if there was a problem communicating with the XRP Ledger.
+    /// - Returns: The status of the given transaction.
+    public func getTx(for transactionHash: TransactionHash) throws -> Rpc_V1_GetTxResponse {
+        let tx = try getRawTx(for: transactionHash)
+        return tx
     }
 
-    guard let signedTransaction = Signer.sign(transaction, with: sourceWallet) else {
-      throw XRPLedgerError.signingError
+    /// Retrieve the transaction status for a given transaction hash.
```
[the remark this answers is not in the document]
Comment (dangell7) [reply XpringKit/DefaultXpringClient.swift]: Can you explain this? Are you referring to the whole doc?

File: XpringKit/DefaultXpringClient.swift [hunk 2]
```diff
@@ -1,146 +1,181 @@
...
-        $0.xrpAmount = xrpAmount
-      }
-      $0.signingPublicKeyHex = sourceWallet.publicKey
-      $0.lastLedgerSequence = lastValidatedLedgerSequence + ledgerSequenceMargin
+    /// Retrieve the transaction status for a given transaction hash.
+    ///
+    /// - Parameter transactionHash: The hash of the transaction.
+    /// - Throws: An error if there was a problem communicating with the XRP Ledger.
+    /// - Returns: The status of the given transaction.
+    public func getTx(for transactionHash: TransactionHash) throws -> Rpc_V1_GetTxResponse {
```
[the remark this answers is not in the document]
Comment (dangell7) [reply XpringKit/DefaultXpringClient.swift]: I see what you mean. Not sure why I just passed them through like that. I'll correct these asap.

**Pull request closed without merging**